op/main.py

```diff
@@ -33,59 +33,92 @@ def _load_metadata(charm_dir):
     return metadata
 
 
-def _handle_event_link(charm_dir, bound_event):
+def _handle_event_link(charm_dir, event_dir, charm_code_link, bound_event):
     """Create a symlink for a particular event.
 
     charm_dir -- A root directory of the charm
     bound_event -- An event for which to create a symlink.
     """
+    # TODO: Handle function/action events here.
+    if not issubclass(bound_event.event_type, op.charm.HookEvent):
+        raise RuntimeError(f'cannot create a symlink: unsupported event type {bound_event.event_type}')
+
     if issubclass(bound_event.event_type, op.charm.InstallEvent):
         # We don't set up the link for install events, since we assume it's already in place
         # (otherwise, we would never have been called).
         return
 
-    event_hook_path = charm_dir / 'hooks' / bound_event.event_kind.replace('_', '-')
+    if not event_dir.exists():
+        raise RuntimeError(f'cannot create event symlink: {event_dir} directory does not exist')
+
+    event_path = event_dir / bound_event.event_kind.replace('_', '-')
     create_link = True
     # Remove incorrect symlinks or files.
-    if event_hook_path.exists():
-        # Non-symlink entries and the ones not pointing to "install"
-        # need to be removed.
-        if not event_hook_path.is_symlink():
-            debugf(f'Hook entry at {event_hook_path} is not a symlink:'
-                   ' attempting to remove it.')
+    if event_path.exists():
+        # Non-symlink entries and the ones not pointing to the charm code file need to be removed.
+        if not event_path.is_symlink():
+            debugf(f'Path at {event_path} is not a symlink: attempting to remove it.')
```

Comment:
Can you remind me why we want to do this?  If the charm author wants to put some non-symlink files in there, why wouldn't we assume that they know what they're doing and leave them alone?  Is this an attempt to preemptively handle upgrades from, say, a reactive charm to this framework?  If so, I wonder if we're adding complexity for a problem that we haven't fully thought through anyway and maybe we should just punt on it for now.

Reply:
OK, we had a discussion with Gustavo during the standup and we can ignore extra files.